gnovm/pkg/gnolang/nodes.go

```diff
@@ -1375,6 +1379,13 @@ func (x *PackageNode) PrepareNewValues(pv *PackageValue) []TypedValue {
 			panic("PackageNode.PrepareNewValues() package mismatch")
 		}
 	}
+	// The FuncValue Body may have been altered during the preprocessing.
+	// We need to update body field from the source in the FuncValue accordingly.
+	for _, tv := range x.Values {
+		if fv, ok := tv.V.(*FuncValue); ok {
+			fv.UpdateBodyFromSource()
```

Comment:
in the place where the body was originally set, add comment that the body may get replaced later.

Reply:
Added in the predefine and try predefine section.  I think these are the only places we set body for FuncValue